src/runtime/components/forms/Rating.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+  <div :class="ui.wrapper">
+    <UIcon
+      v-for="star of [...Array(max).keys()].slice().reverse()"
+      :key="star"
+      :name="icon"
+      :class="[
+        ratingClass,
+        (star + 1 <= modelValue ? ui.active : ui.inactive).replaceAll(
+          '{color}',
+          color
+        ),
+        readOnly ? 'cursor-auto hover:text-gray-300 hover:dark:text-gray-700 peer-hover:text-gray-300 peer-hover:dark:text-gray-700' : 'cursor-pointer',
+        'order-{order}'.replaceAll('{order}', star.toString()),
+      ]"
```

Comment:
Can be simplify to this

1. add `readOnly` to `ratingClass`
2. use `rate` not `modelValue` => `rate >= star + 1` and make it in one line

```js
:class="[
  ratingClass,
  (rate >= star + 1 ? ui.active : ui.inactive).replaceAll('{color}',color),
  'order-{order}'.replaceAll('{order}', star.toString()),
 ]"
```

 add `readOnly` to `ratingClass`

```js
const ratingClass = computed(() => {
  return classNames(
    ui.value.base.replaceAll('{color}', props.color),
    ui.value.size[props.size],
    props.readOnly ?  'cursor-auto hover:text-gray-300 hover:dark:text-gray-700 peer-hover:text-gray-300 peer-hover:dark:text-gray-700' : 'cursor-pointer'
  )
})
```

Reply:
@larrasu, really, it's a great component and great work!